mmocr/apis/inferencers/kie_inferencer.py

```diff
@@ -53,63 +46,188 @@ def _init_pipeline(self, cfg: ConfigType) -> None:
         # If it's in non-visual mode, self.pipeline will be specified.
         # Otherwise, file_pipeline and ndarray_pipeline will be specified.
         if self.novisual:
-            self.pipeline = Compose(pipeline_cfg)
-        else:
-            return super()._init_pipeline(cfg)
-
-    def _init_visualizer(self, cfg: ConfigType) -> None:
-        """Initialize visualizers."""
-        # TODO: We don't export images via backends since the interface
-        # of the visualizer will have to be refactored.
-        self.visualizer = None
-        if 'visualizer' in cfg:
-            self.visualizer = VISUALIZERS.build(cfg.visualizer)
-            dataset = DATASETS.build(cfg.test_dataloader.dataset)
-            self.visualizer.dataset_meta = dataset.metainfo
-
-    def preprocess(self, inputs: InputsType) -> List[Dict]:
-        results = []
+            return Compose(pipeline_cfg)
+        return super()._init_pipeline(cfg)
+
+    def _init_visualizer(self, cfg: ConfigType) -> Optional[Visualizer]:
+        """Initialize visualizers.
+
+        Args:
+            cfg (ConfigType): Config containing the visualizer information.
+
+        Returns:
+            Visualizer or None: Visualizer initialized with config.
+        """
+        visualizer = super()._init_visualizer(cfg)
+        dataset = DATASETS.build(cfg.test_dataloader.dataset)
+        visualizer.dataset_meta = dataset.metainfo
```

Comment:
metainfo can be obtained from the model instead of the dataset.
The dataset should not be necessary for inference